### ISSUE TITLE
Request: No Flashy Equipment

### ISSUE THREAD

**Issue by Rydian** [cursey/kanan-new#34]
The following signature...
`8B 10 89 57 1C 8B 50 04`
Gets you to a block of code that copies/manages color codes for item rendering (in CEquipmentItemInfo).  Specifically, the following three instructions (just below where the signature lands) deal with the three colors, offsets 0x1C, 0x20, and 0x24.
```
mov [edi+1C],edx
[...]
mov [edi+20],edx
[...]
mov [edi+24],eax
```

Color codes in Mabinogi are 4 bytes, MMRRGGBB, where the MM byte is a meta/flashy byte.  A value of 00 or FF for the MM byte means not flashy (00 tends to mean generated while FF tends to be gacha/event reward script dyes with hand-typed codes for those that are curious but both have the same effect in-game).

By overwriting the MM bytes to 00 in the code that copies item data for rendering, we can prevent flashy items from flashing when worn (they stick to the first color), while still leaving them flashing in the inventory (so players don't accidentally use a flashy dye on their outfit or something).

... problem is the hair system.  Hair is a worn item, and the color code on hair is XX000000, where XX is a color ID defined in various databases (opposed to being a hex color code).  Simply overwriting the first by to 00 in all cases makes hair come out all black (as it turns to 00000000 and color ID 0 is black).

Additional code needs to be inserted to check for the XX000000 format and skip the overwrite if it looks like a hair ID.  In my CE table (and old Kanan) I did the check/skip with with a bunch of custom ASM, but it should be less annoying for somebody better at coding to do it in C++.

**Comment by cursey** [comment]
Sorry for the long delay getting around to this. I read through your post and uhh this should be the easiest thing to add ever because of how Equipment Override is implemented. I'll attempt adding it now.

**Comment by cursey** [comment]
Implemented in commit 80afb45cdc1d70bf2dc6e8c6bd7ac09dc82ac787
